tests/ctst/steps/bucket-policies/utils.ts

```diff
@@ -0,0 +1,500 @@
+type ActionPermissionsType = {
+    action: string;
+    permissions: string[];
+    subAuthorizationChecks?: boolean;
+    expectedResultOnAllowTest?: string;
+    excludePermissionOnBucketObjects?: boolean,
+    excludePermissionsOnBucket?: boolean,
+    useWildCardBucketName?: boolean,
+    needsSetup?: boolean,
+};
+
+const needObjectLock = [
+    'GetObjectLegalHold',
+    'PutObjectLegalHold',
+    'GetObjectRetention',
+    'PutObjectRetention',
+    'PutObjectVersionLegalHold',
+    'PutObjectVersionRetention',
+    'GetObjectVersionLegalHold',
+    'GetObjectVersionRetention',
+];
+
+const needObject = [
+    'PutObjectLegalHold',
+    'PutObjectRetention',
+    'PutObjectTagging',
+    'PutObjectAcl',
+    'PutObject',
+    'GetObject',
+    'GetObjectAcl',
+    'GetObjectLegalHold',
+    'GetObjectRetention',
+    'GetObjectTagging',
+    'DeleteObject',
+    'DeleteObjectTagging',
+    'DeleteObjects',
+    'HeadObject',
+    'UploadPart',
+    'UploadPartCopy',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const needVersioning = [
+    'PutBucketReplication',
+    'ListObjectVersions',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const actionPermissions: ActionPermissionsType[] = [
+    // Create bucket is not checking any bucket policy
+    // {
+    //     action: 'CreateBucket',
+    //     permissions: ['s3:CreateBucket'],
+    //     expectedResultOnAllowTest: 'BucketAlreadyOwnedByYou',
+    //     excludePermissionOnBucketObjects: true,
+    //     useWildCardBucketName: true,
+    // },
+    {
+        action: 'PutBucketAcl',
+        permissions: ['s3:PutBucketAcl'],
+        expectedResultOnAllowTest: 'MalformedXML',
```

Comment:
I don't see the same expected result in Integration: https://github.com/scality/Integration/blob/development/7.10/tests/bucket-policy/utils/s3APIConfigurations.js#L98

Reply:
Two things are different in CTST as compared with Integration:
- The `expectedResultOnAllowTest` is checked if there is an error, if there is no error, it passes if the scenario is expecting an `Allow`
- This error, for this specific API is returned [after the authz](https://github.com/scality/cloudserver/blob/2596f3fda8265b981e0241dcf3a66a87379efb79/lib/api/bucketPutACL.js#L95), so we can keep it

For this one we do get this error, because we send an empty body, which is fine, we are not testing the full API here
```
2024-02-21T13:34:54.7723066Z {"reqId":1865,"api":"PutBucketAclCommand","input":{"Bucket":"zenko-ctstbucketfortesthxs5u"},"err":"MalformedXML: The XML you provided was not well-formed or did not validate against our published schema.","accessKey":"C3FIRAPABNW8H4LPX8FS","secretKey":"/02sXjoOgje/tjG0rn5FtRjhwZ7d=d8Pg0Q5da30","isSessionToken":true,"endpoint":"http://s3.zenko.local:80"}
```